src/test/java/com/kodilla/ecommercee/domain/ProductTestSuite.java

```diff
@@ -0,0 +1,163 @@
+        Product testProduct = new Product();
+        testProduct.setName("Shirt");
+        testProduct.setDescription("white shirt");
+        testProduct.setPrice(BigDecimal.valueOf(5.20));
+
+        //when
+        productRepository.save(testProduct);
+
+        //then
+        Assert.assertEquals(1L, productRepository.count());
+    }
+
+    @Transactional
+    @Test
+    public void testFindById() {
+        //given
+        Product testProduct = new Product("Shirt", "white shirt", BigDecimal.valueOf(5.20));
+
+        //when
+        productRepository.save(testProduct);
+        Long id = testProduct.getId();
+        Optional<Product> loadedProduct = productRepository.findById(id);
+
+        //then
+        Assert.assertTrue(loadedProduct.isPresent());
+        Assert.assertEquals("Shirt", loadedProduct.get().getName());
+        Assert.assertEquals(BigDecimal.valueOf(5.20), loadedProduct.get().getPrice());
+
+    }
+
+    @Transactional
+    @Test
+    public void testRelationWithGroups() {
+        //given
+        Group clothes = new Group();
+        Group food = new Group();
+
+        groupRepository.save(clothes);
+        groupRepository.save(food);
+
+        Product trousers = new Product("trousers", "normal trousers", BigDecimal.valueOf(22.90));
+        trousers.setGroup(clothes);
+
+        Product tomato = new Product("tomato", "red tomato", BigDecimal.valueOf(0.99));
+        tomato.setGroup(food);
+
+        productRepository.save(trousers);
+        productRepository.save(tomato);
+
+        Long clothesId = clothes.getId();
+        Long foodId = food.getId();
+
+        Long trousersId = trousers.getId();
+        Long tomatoId = tomato.getId();
+
+        //when
+        Product loadedTrousers = productRepository.findById(trousersId).get();
+        Long loadedTrousersGroupId = loadedTrousers.getGroup().getId();
+
+        Product loadedTomato = productRepository.findById(tomatoId).get();
+        Long loadedTomatoGroupId =loadedTomato.getGroup().getId();
+
+        //then
+        Assert.assertEquals(clothesId, loadedTrousersGroupId);
+        Assert.assertEquals(foodId, loadedTomatoGroupId);
+    }
+
+    @Transactional
+    @Test
+    public void testFindAll() {
+        //given
+        Product testProductOne = new Product("Shirt", "white shirt", BigDecimal.valueOf(5.20));
+        Product testProductTwo = new Product("Hat", "fancy hat", BigDecimal.valueOf(8.99));
+        Product testProductThree = new Product("Pants", "panties", BigDecimal.valueOf(3.00));
+        Product testProductFour = new Product("Scarf", "nice scarf", BigDecimal.valueOf(7.05));
+
+        productRepository.save(testProductOne);
+        productRepository.save(testProductTwo);
+        productRepository.save(testProductThree);
+        productRepository.save(testProductFour);
+
+        //when
+        List<Product> loadedProducts = productRepository.findAll();
+
+        //then
+        Assert.assertEquals(4, loadedProducts.size());
```

Comment:
To samo co w pierwszym teście.

Reply:
Poprawione :) Teraz nie sprawdza już ilości, tylko występowanie konkretnych produktów